packages/@ember/routing/route.ts

```diff
@@ -1400,12 +1401,20 @@ class Route<T = unknown>
     if a promise returned from `model` fails, the error will be
     handled by the `error` hook on `Route`.
 
+    Note that the legacy behavior of automatically defining a model
+    hook when a dynamic segment ending in `_id` is present is deprecated.
+    You should explicitly define a model hook whenever any segments are
+    present.
```

Comment:
Add deprecation id or 6.0 mention here so we make sure to remove this in the future.